qiskit_experiments/tomography/tomography_experiment.py

```diff
@@ -0,0 +1,307 @@
+from .basis import BaseTomographyMeasurementBasis, BaseTomographyPreparationBasis
+from .tomography_analysis import TomographyAnalysis
+
+
+class TomographyExperiment(BaseExperiment):
+    """Base experiment for quantum state and process tomography"""
+
+    __analysis_class__ = TomographyAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls):
+        return Options(basis_indices=None)
+
+    def __init__(
+        self,
+        circuit: Union[QuantumCircuit, Instruction, BaseOperator],
+        measurement_basis: Optional[BaseTomographyMeasurementBasis] = None,
+        measurement_qubits: Optional[Iterable[int]] = None,
+        preparation_basis: Optional[BaseTomographyPreparationBasis] = None,
+        preparation_qubits: Optional[Iterable[int]] = None,
+        basis_indices: Optional[Iterable[Tuple[List[int], List[int]]]] = None,
+        qubits: Optional[Iterable[int]] = None,
+    ):
+        """Initialize a state tomography experiment.
+
+        Args:
+            circuit: the quantum process circuit. If not a quantum circuit
+                it must be a class that can be appended to a quantum circuit.
+            measurement_basis: Tomography basis for measurements.
+            measurement_qubits: Optional, the qubits to be measured. These should refer
+                to the logical qubits in the state circuit.
+            preparation_basis: Tomography basis for measurements.
+            preparation_qubits: Optional, the qubits to be prepared. These should refer
+                to the logical qubits in the process circuit.
+            basis_indices: Optional, the basis elements to be measured. If None
+                All basis elements will be measured.
+            qubits: Optional, the physical qubits for the initial state circuit.
+        """
+        # Initialize BaseExperiment
+        if qubits is None:
+            qubits = circuit.num_qubits
+        super().__init__(qubits)
+
+        # Get the target tomography circuit
+        if isinstance(circuit, QuantumCircuit):
+            target_circuit = circuit
+        else:
+            # Convert input to a circuit
+            num_qubits = circuit.num_qubits
+            target_circuit = QuantumCircuit(num_qubits)
+            target_circuit.append(circuit, range(num_qubits))
+        self._circuit = target_circuit
+
+        # Measurement basis and qubits
+        self._meas_circ_basis = measurement_basis
+        if measurement_qubits:
+            self._meas_qubits = tuple(measurement_qubits)
+        else:
+            self._meas_qubits = None
+
+        # Preparation basis and qubits
+        self._prep_circ_basis = preparation_basis
+        if preparation_qubits:
+            self._prep_qubits = tuple(preparation_qubits)
+        else:
+            self._prep_qubits = None
+
+        # Configure experiment options
+        if basis_indices:
+            self.set_experiment_options(basis_indices=basis_indices)
+
+        # Compute target state
```

Comment:
Can you elaborate this in comment? I guess this is just used for computation of the fidelity in analysis. I was wondering why we need this when I first saw this.

Reply:
Yes its just use for fidelity analysis